eds: refactor tests to init clusters and endpoints relevant for test only

ramaraochavali opened the pull request:
Fixes https://github.com/istio/istio/issues/20343

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 0cdf20e51f90b9b13b05ace3c81cd61caaff2ca1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20403/unit-tests_istio/7036) | `/test unit-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/test unit-tests_istio

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 0cdf20e51f90b9b13b05ace3c81cd61caaff2ca1 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20403/unit-tests_istio/7039) | `/test unit-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

istio-testing commented:
@ramaraochavali: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
unit-tests_istio | 9a597bf878c40ad7874db59b0acb3b13818503c8 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_istio/20403/unit-tests_istio/7040) | `/test unit-tests_istio`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

ramaraochavali commented:
/retest

ramaraochavali commented on pilot/pkg/proxy/envoy/v2/debug_test.go:
```diff
@@ -315,7 +315,7 @@ func TestAuthenticationZ(t *testing.T) {
 			proxyID:        "dumpApp-644fc65469-96dza.testns",
 			wantCode:       200,
-			expectedLength: 25,
+			expectedLength: 24,
```
This change is because now I moved edsInc services out of standard init to the test, which creates a endpoint with ServiceAccount

ramaraochavali commented:
@howardjohn PTAL. I made couple of changes
- Introduced an init function so that we only initialize the clusters and endpoints needed for the test, not the entire world
- Adjusted some timeouts.

This has reduced the time takes to run eds tests from 26s to close to 13s .

However, I think still there is a genuine race condition in SetDestinationLabels here https://github.com/istio/istio/blob/master/pilot/pkg/model/push_context.go#L1402 where we modify host, which we will have decide whether it needs to be addressed or we can live with it.

howardjohn commented:
Ah we had that same race for VS as well, I think we cloned them or
something to fix it. I'll take a closer look

On Wed, Jan 22, 2020 at 12:26 AM Rama Chavali <notifications@github.com>
wrote:

> @howardjohn <https://github.com/howardjohn> PTAL. I made couple of changes
>
>    - Introduced an init function so that we only initialize the clusters
>    and endpoints needed for the test, not the entire world
>    - Adjusted some timeouts.
>
> This has reduced the time takes to run eds tests from 26s to close to 13s .
>
> However, I think still there is a genuine race condition in
> SetDestinationLabels here
> https://github.com/istio/istio/blob/master/pilot/pkg/model/push_context.go#L1402
> where we modify host, which we will have decide whether it needs to be
> addressed or we can live with it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/istio/istio/pull/20403?email_source=notifications&email_token=AAEYGXK5O7M5Z2KQUK5G2VDQ677KJA5CNFSM4KJ7TWW2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEJSUWDQ#issuecomment-577063694>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEYGXJVJVJZVL7ZCNA3YK3Q677KJANCNFSM4KJ7TWWQ>
> .
>
